services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/application/ApplicationData.kt

```diff
@@ -96,6 +96,20 @@ data class CableReportApplicationData(
         customersWithContacts().flatMap { it.contacts }.find { it.orderer }
 }
 
+/**
+ * An extension function to get email addresses from customer contact persons. Returns a set of
+ * emails that:
+ * - are not null, empty or blank.
+ * - do not match the optional [omit] argument.
+ */
+fun ApplicationData.contactPersonEmails(omit: String? = null): Set<String> =
+    when (this) {
+        is CableReportApplicationData ->
+            customersWithContacts().flatMap { it.contactPersonEmails(omit) }.toSet()
+    }
```

Comment:
Couldn't this be replaced with a function in the interface? The interface already says all implementations need to have a `customersWithContacts` field, so the type-checking doesn't seem necessary.
```
sealed interface ApplicationData {
...
    fun contactPersonEmails(omit: String? = null): Set<String> =
        customersWithContacts().flatMap { it.contactPersonEmails(omit) }.toSet()
}
```

Reply:
That is a good point. I moved it to the interface as default function.